app/src/main/java/com/mailerdaemon/app/LoginActivity.kt

```diff
@@ -0,0 +1,175 @@
+package com.mailerdaemon.app
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import android.os.Bundle
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import com.facebook.AccessToken
+import com.facebook.CallbackManager
+import com.facebook.FacebookCallback
+import com.facebook.FacebookException
+import com.facebook.login.LoginManager
+import com.facebook.login.LoginResult
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import com.google.android.gms.common.api.ApiException
+import com.google.android.gms.tasks.Task
+import com.google.firebase.auth.*
+import com.google.firebase.firestore.FirebaseFirestore
+import kotlinx.android.synthetic.main.activity_login.*
+import java.util.*
+
+class LoginActivity: AppCompatActivity() {
+
+    private val email = "email"
+    private val rcsignin = 234
+    private lateinit var mAuth: FirebaseAuth
+    private lateinit var callbackManager: CallbackManager
+    
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setTheme(R.style.AppTheme_NoActionBar_NoStatusColor)
+        val mAuth = FirebaseAuth.getInstance()
+        if (getSharedPreferences(MAIN, Context.MODE_PRIVATE).getBoolean("intro", true)) {
+            startActivity(Intent(this, IntroActivity::class.java))
+            finish()
+        } else
+            startMain()
+        setContentView(R.layout.activity_login)
+        progress_bar.visibility = View.GONE
+        forgot_password.setOnClickListener { startActivity(Intent(this, ForgotPassActivity::class.java)) }
+        login.setOnClickListener {
+            val email = login_email.text.toString().trim { it <= ' ' }
+            val password = (login_password.text).toString()
+            if (email.isNotEmpty()) {
+                if (password.isNotEmpty()) {
+                    progress_bar.visibility = View.VISIBLE
+                    mAuth.signInWithEmailAndPassword(email, password).addOnCompleteListener { task: Task<AuthResult?> ->
+                        progress_bar.visibility = View.GONE
+                        if (task.isSuccessful)
+                            saveUser((mAuth.currentUser))
+                         else
+                            this.toast("Invalid password")
+                    }
+                } else login_password.error = "Password cannot be empty"
```

Comment:
Extract this String as String Resources. All the UI messages should use string resource instead of hard coded text.